frontend/src/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,33 @@
+import { User, onAuthStateChanged } from "firebase/auth";
+import React, { ReactNode, useEffect, useMemo, useState } from "react";
+
+import { auth } from "@/firebase";
+
+type ProviderProps = {
+  children: ReactNode;
+};
+
+type contextType = {
+  currentUser: User | null;
+  signedIn: boolean;
+};
+
+export const AuthContext = React.createContext<contextType>({ currentUser: null, signedIn: false });
+
+export function AuthProvider({ children }: ProviderProps) {
+  const [currentUser, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    onAuthStateChanged(auth, (user: User | null) => {
+      if (user) {
+        setUser(user);
+      } else {
+        setUser(null);
+      }
+    });
+  });
```

Comment:
Just as good practice, it is a good idea to [return a cleanup function](https://blog.logrocket.com/understanding-react-useeffect-cleanup-function/) ([see example](https://github.com/TritonSE/TAP-Foster-Dog-Matching/blob/101fb04c11dd45d90e6956a7c417414006bbafe4/client/src/contexts/AuthContext.js#L50-L70)) so that the listener is removed after the component is destroyed. Doing this prevents dangling listeners (and keeps the app free of memory-leaks). Also, currently, the code is registering a new listener every time the component rerenders (we only want a listener to get registered once). Passing `[]` as the dependencies for the effect will ensure the effect runs only once (when the component renders for the first time).

```suggestion
  useEffect(() => {
    const unsubscribe = onAuthStateChanged(auth, (user: User | null) => {
      if (user) {
        setUser(user);
      } else {
        setUser(null);
      }
    });
    return unsubscribe;
  }, []);
```

Reply:
Fixed all of this in my NavBar pr, forgot to change it here too, sorry about that. Would it be fine to leave this as is and have it be fixed by the nav pr? (I made a lot of other changes to the file in that pr too)

Reply:
Oh gotcha. Just saw that lol. Yep, that works. Feel free to ignore this comment then.